### PR TITLE
RD 10961 + RD 10948 fixes

### PR DESCRIPTION
**Pull request by Alexzerntev**
- Added support for multiline strings and quoted identifiers
- Fixed the position out of grid crash

### CODE REVIEW

**Comment by bgaidioz** [comment]
This test:
```scala
test("Test multiple param occurrences")
```
fails. This is new. I see the test code contains an extra `"` in the end (the triple quoted string containing the query ends with `""""`). I imagine it's due to that?

File: sql-client/src/test/scala/raw/client/sql/TestSqlCompilerServiceAirports.scala [hunk 1]
```diff
@@ -963,35 +963,35 @@ class TestSqlCompilerServiceAirports
...
+  test("""RD-10948+10961""") { _ =>
     val q = """:
-              |""".stripMargin
+      |""".stripMargin
     val ValidateResponse(errors) = compilerService.validate(q, asJson())
     assert(errors.nonEmpty)
   }
 
-  test("""RD-10948"""){_ =>
+  test("""RD-10948""") { _ =>
```
[the remark this answers is not in the document]
Comment (Alexzerntev) [reply sql-client/src/test/scala/raw/client/sql/TestSqlCompilerServiceAirports.scala:973]: I alternated it as we discussed

File: sql-client/src/test/scala/raw/client/sql/TestSqlCompilerServiceAirports.scala [hunk 2]
```diff
@@ -963,35 +963,35 @@ class TestSqlCompilerServiceAirports
...
+      |
+      |SELECT * FROM
+      |(VALUES
+      |  (1, 'John', 'Doe', DATE '2023-01-01'),
+      |  (2, 'Jane', 'Doe', DATE '2024-01-01')
+      |) as i(id, first_name, last_name, birthday)
+      |WHERE id = :id && id = :
+      |""".stripMargin
     val ValidateResponse(errors) = compilerService.validate(q, asJson())
     assert(errors.nonEmpty)
```
[the remark this answers is not in the document]
Comment (Alexzerntev) [reply sql-client/src/test/scala/raw/client/sql/TestSqlCompilerServiceAirports.scala:996]: No errors now

[the remark this answers is not in the document]
Comment (Alexzerntev) [reply sql-client/src/test/scala/raw/client/sql/TestSqlCompilerServiceAirports.scala:996]: syntax error at or near "="
Is this error expected?

**Pull request merged**